File: facebook-messenger/discover/java/app/src/main/java/ai/codia/x/java/demo/MainActivity.java
package ai.codia.x.java.demo;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class MainActivity extends BaseMainActivity {

    @Override
    protected void onViewCreated() {
        
    }

    private int dp2px(double dpValue) {
        float scale = this.getResources().getDisplayMetrics().density;
        return (int) (dpValue * scale);
    }
}

File: facebook-messanger/discover/kotlin/app/src/main/java/ai/codia/x/kotlin/demo/MainActivity.kt
package ai.codia.x.kotlin.demo

import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : BaseMainActivity() {

    override fun onViewCreated() {
        
    }

    private fun dp2px(dpValue: Double): Int {
        val scale = this.resources.displayMetrics.density
        return (dpValue * scale).toInt()
    }
}
